ocaml/typing/types.ml

```diff
@@ -668,7 +668,7 @@ let log_change ch =
   trail := r'
 
 let () =
-  Mode.change_log := (fun changes -> log_change (Cmodes changes));
+  Mode.append_changes := (fun changes -> log_change (Cmodes !changes));
```

Comment:
This is the only mention of `append_changes` outside the solver infrastructure. Instead of exposing the whole `ref`, I think it's better to expose a `set_append_changes` that writes the mutable variable. There's no need for any code outside the solver to access the value of this variable.